mlflow/server/auth/__init__.py

```diff
@@ -8,15 +8,59 @@
 """
 
 import logging
+import uuid
+import os
+from pathlib import Path
+from flask import Flask, request, make_response, Response, redirect, flash, render_template_string
 
 from mlflow.server import app
+from mlflow.server.auth.config import read_auth_config
+from mlflow.server.auth.sqlalchemy_store import SqlAlchemyStore
+from mlflow.server.handlers import _get_rest_path
 
+_AUTH_CONFIG_PATH_ENV_VAR = "MLFLOW_AUTH_CONFIG_PATH"
 
 _logger = logging.getLogger(__name__)
 
+store = SqlAlchemyStore()
+
+
+class ROUTES:
+    USERS = _get_rest_path("/mlflow/users")
+    SIGNUP = _get_rest_path("/mlflow/signup")
+
+
+UNPROTECTED_ROUTES = [ROUTES.USERS, ROUTES.SIGNUP]
+
+
+def is_unprotected_route(path: str) -> bool:
+    if path.startswith("/static"):
+        return True
+    return path in UNPROTECTED_ROUTES
+
+
+def make_basic_auth_response() -> Response:
+    res = make_response()
+    res.status_code = 401
+    res.headers["WWW-Authenticate"] = 'Basic realm="mlflow"'
+    return res
+
 
 def _before_request():
-    # TODO: Implement authentication
+    _user = request.authorization.username if request.authorization else None
+    _logger.debug(f"before_request: {request.method} {request.path} (user: {_user})")
+    if is_unprotected_route(request.path):
+        return
```

Comment:
```suggestion
    if is_unprotected_route(request.path):
        return
    _user = request.authorization.username if request.authorization else None
    _logger.debug(f"before_request: {request.method} {request.path} (user: {_user})")

```

Can we return earlier?

Reply:
updated